tests/semantic_asserts.py

```diff
@@ -0,0 +1,299 @@
+from typing import Callable, Dict, List, Union
+
+from algosdk import kmd
+from algosdk.v2client import algod, indexer
+
+import algosdk.testing.teal_blackbox as blackbox
+
+from pyteal import *
+
+## Clients
+
+
+def _algod_client(
+    algod_address="http://localhost:4001", algod_token="a" * 64
+) -> algod.AlgodClient:
+    """Instantiate and return Algod client object."""
+    return algod.AlgodClient(algod_token, algod_address)
+
+
+def _kmd_client(
+    kmd_address="http://localhost:4002", kmd_token="a" * 64
+) -> kmd.KMDClient:
+    """Instantiate and return a KMD client object."""
+    return kmd.KMDClient(kmd_token, kmd_address)
+
+
+def _indexer_client(
+    indexer_address="http://localhost:8980", indexer_token="a" * 64
+) -> indexer.IndexerClient:
+    """Instantiate and return Indexer client object."""
+    return indexer.IndexerClient(indexer_token, indexer_address)
+
+
+def algod_with_assertion():
+    algod = _algod_client()
+    assert algod.status(), "alod.status() did not produce any results"
```

Comment:
@tzaffi Friendly pointer to minor typo.

```suggestion
    assert algod.status(), "algod.status() did not produce any results"
```